src/mpl/src/gpu/mpl_gpu_cuda.c

```diff
@@ -309,12 +309,11 @@ int MPL_gpu_init(int debug_summary)
     }
 
     cudaError_t ret = cudaGetDeviceCount(&device_count);
-    CUDA_ERR_CHECK(ret);
-
-    if (device_count <= 0) {
+    if (ret == cudaErrorNoDevice || (ret == cudaSuccess && device_count <= 0)) {
```

Comment:
Don't you have to consume the error code with `cudaGetLastError()` in case you get cudaErrorNoDevice?

Reply:
Thanks, I wasn't aware of this need. Will add.